tests/tests_unit/test_credential_providers.py

```diff
@@ -52,9 +54,61 @@ def test_token_non_string(self):
         ):
             Token({"foo": "bar"})
 
-    def test_create_from_credential_provider(self):
-        creds = CredentialProvider.load({"token": "abc"})
+    @pytest.mark.parametrize(
+        "config",
+        [
+            {"token": "abc"},
+            {"token": {"token": "abc"}},
+            '{"token": "abc"}',
+            '{"token": {"token": "abc"}}',
+            {"token": (lambda: "abc")},
+            {"token": {"token": (lambda: "abc")}},
+        ],
+    )
+    def test_create_from_credential_provider(self, config):
+        creds = CredentialProvider.load(config)
         assert isinstance(creds, Token)
+        assert "Authorization", "Bearer abc" == creds.authorization_header()
```

Comment:
Add another `Token.load` (maybe in the same test?) Could add a flag like "accepted"..